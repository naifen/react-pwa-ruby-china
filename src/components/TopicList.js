import React from "react";
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';
import Divider from '@material-ui/core/Divider';
import Button from '@material-ui/core/Button';
import ImageIcon from '@material-ui/icons/Image';
import TopAppBar from './TopAppBar';

const styles = theme => ({
  root: {
    width: '100%',
    overflow: 'auto',
    paddingBottom: 60,
    backgroundColor: theme.palette.background.default,
  },
});

// TODO: hide TopAppBar while scroll down, show while scroll up
const TopicList = ({ classes, topics }) => (
  <div>
    <TopAppBar />
    <Paper className={classes.root} elevation={1}>
      <List>
        {topics.map(topic => (
          <React.Fragment key={topic.id}>
              <ListItem>
                <Avatar>
                  <ImageIcon />
                </Avatar>
                <ListItemText primary={topic.title} secondary="Author Jan 9, 2014" />
              </ListItem>
              <Divider light />
          </React.Fragment>
        ))}
      </List>

      <Button variant="contained" color="primary">
        Load more
      </Button>
    </Paper>
	</div>
);

TopicList.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(TopicList);
