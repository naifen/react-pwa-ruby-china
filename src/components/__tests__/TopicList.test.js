import React from "react";
import { shallow } from "enzyme";
import TopicList from "../TopicList";

describe("<TopicList />", () => {
  it("should render correctly with no props", () => {
    const component = shallow(<TopicList />);
    expect(component).toMatchSnapshot();
  });
  it("should render banner text correctly with given strings", () => {
    const component = shallow(<TopicList classes={{}} />);
    expect(component).toMatchSnapshot();
  });
});
