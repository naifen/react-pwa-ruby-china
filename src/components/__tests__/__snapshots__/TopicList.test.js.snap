// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicList /> should render banner text correctly with given strings 1`] = `
<TopicList
  classes={
    Object {
      "progress": "TopicList-progress-2",
      "root": "TopicList-root-1",
    }
  }
/>
`;

exports[`<TopicList /> should render correctly with no props 1`] = `
<TopicList
  classes={
    Object {
      "progress": "TopicList-progress-2",
      "root": "TopicList-root-1",
    }
  }
/>
`;
